lib/checkPullRequestTemplate.js

```diff
@@ -165,30 +166,78 @@ const checkEssentialChecklist = (pullRequest) => {
         const isPrFromFork =
           pullRequest.head.repo.full_name === pullRequest.base.repo.full_name;
         if (!pullRequest.maintainer_can_modify && !isPrFromFork) {
-          if (message) {
-            message +=
-              'The allow edits from maintainers checklist needs to ' +
-              'be ticked so that maintainers can rerun failed tests. ' +
-              'Endeavour to add this by ticking on the check box. ';
-          } else {
-            message +=
-              'the allow edits from maintainers checklist needs to ' +
-              'be ticked so that maintainers can rerun failed tests. ' +
-              'Endeavour to add this by ticking on the check box. ';
-          }
+          message +=
+            'The allow edits from maintainers checklist needs to ' +
+            'be ticked so that maintainers can rerun failed tests. ' +
+            'Endeavour to add this by ticking on the check box. ';
         }
       }
     });
 
     return message;
   } else {
     return (
-      'the body of this PR is missing the checklist section, please ' +
+      'The body of this PR is missing the checklist section, please ' +
       'update it to include the checklist.'
     );
   }
 };
 
+
+/**
+ * This function checks if the pull request's body contains the
+ * checklist and returns an appropriate message if the required
+ * checklist items are not checked.
+ *
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const checkProofThatChangesAreCorrect = (pullRequest) => {
+  /*
+    A valid Proof that changes are correct section would look like:
+    ## Proof that changes are correct
+
+    <!--
+    Add videos/screenshots of the user-facing interface to demonstrate
+    that the changes made in this PR work correctly.
+    -->
+    ![oppiabot-test](https://user-images.githubusercontent.com/30050862/
+      test.gif)
+    (or)
+    https://user-images.githubusercontent.com/30050862/
+      test.mp4
+      */
+  const proofOfChcangesHeadingIndex = pullRequest.body.indexOf(PROOF_TEXT);
+
+  if (proofOfChcangesHeadingIndex !== -1) {
+    const nextHeadingIndex = pullRequest.body.indexOf(
+      HEADING_START,
+      proofOfChcangesHeadingIndex + PROOF_TEXT.length
+    );
+
+    let proofOfChangesSection = pullRequest.body.substring(
+      proofOfChcangesHeadingIndex,
+      nextHeadingIndex
+    );
+
+    var proofLinks = proofOfChangesSection.includes(UPLOADED_MEDIA);
```

Comment:
Hi, I think the problem with this is that sometimes the devs might want to upload pictures in another way than directly via GitHub, for example, a link to the images or videos. So I think it would be enough to check if there is at least something in the section.

Reply:
@vojtechjelinek could we check for the presence of `http` in that section instead?
Reason:
1. Even an enter or one new character can look like some txt is present.
2. a link will contain a `http` in the beginning of it.

Reply:
I tried to check for the presence of other txt by checking if the number of characters of in that section is greater than the number of characters present in the same section in the PR template

Reply:
I think we should check that a valid link is provided.

Reply:
So, do I check for the presence of http? or try a regex to validate the whole url?

Reply:
Yep checking for the presence of http or https is a smart solution, I'm fine with either checking just for 'http' or for a full URL. 

Also, I just realized that for some changes for example to backend or typing there is no proof of changes needed, so in taht case we should check if the section starts with something like 'No proof of changes needed because' and also edit the PR template to include this info so that people know they always need to fill the section.

Reply:
> also edit the PR template to include this info so that people know they always need to fill the section.

```
<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes made in this PR work correctly.
-->
```
The above is already present in the Proof that changes are correct section.

Should we include a point in the PR Pointers section to tell to add `No proof of changes needed because` if no proof are present?

Reply:
I would add that directly to the section you mention above.

Reply:
Implemented. Will add Tests after confirmation of approach